naga/src/front/wgsl/lower/mod.rs

```diff
@@ -372,14 +372,29 @@ pub struct ExpressionContext<'source, 'temp, 'out> {
 }
 
 impl<'source, 'temp, 'out> ExpressionContext<'source, 'temp, 'out> {
+    #[allow(dead_code)]
     fn as_const(&mut self) -> ExpressionContext<'source, '_, '_> {
```

Comment:
It might make sense to remove this and add it back later.

Reply:
I agree; the PR would become a one line change then.

I think it would be fine to fold it into https://github.com/gfx-rs/wgpu/pull/6935. Or would you still like to land this separately?

Reply:
I think it would make sense to land it as it stands [now](https://github.com/gfx-rs/wgpu/pull/6945/commits/e547ee342f995dd3c366fd765f800e8a4302f35c).

Reply:
Or maybe not given how clippy/rustdoc hates me, but I would still like to land this separately. Reverted to initial version.